runners/core-java/src/main/java/org/apache/beam/runners/core/triggers/AfterSynchronizedProcessingTimeStateMachine.java

```diff
@@ -27,13 +27,17 @@
 
 class AfterSynchronizedProcessingTimeStateMachine extends AfterDelayFromFirstElementStateMachine {
```

Comment:
Yea, I was just trying to keep this PR manageably sized. Since the state machines are basically independent of the proto concerns, I'd prefer not to mess with them. There will certainly be a follow-up PR where a state machine is built from a proto instead of an SDK trigger. It is not mandatory, since we could parse the proto to an SDK trigger then go from there to a state machine, but it seems more faithful to the model. And it will be net negative LOC. All that is to say that when I say "save it for later" it is a real time that will come, not an excuse to not do it.